File: csharp/LockController/LockControllerLib/IMassiveDbConnection.cs
namespace LockControllerLib;

public interface IMassiveDbConnection
{
    IMassiveSession Session { get; set; }
    MassiveLocation getPathTo(string mObject);
    IMassiveObject FullObject(IMassiveLocation path);
    IMassiveLocation Wangle(IMassiveObject fullMObject);

    int CallMassiveMethod(
                IMassiveLocation path, 
                IMassiveLocation wangleLocation, 
                List<IMassiveVariant> inParameters,
                List<IMassiveVariant> outParameters, 
                List<string> errors);
}

File: typescript/src/IMassiveDbConnection.ts
import { IMassiveLocation, MassiveLocation } from './IMassiveLocation';
import { IMassiveObject } from './IMassiveObject';
import { IMassiveSession } from './IMassiveSession';
import { IMassiveVariant } from './IMassiveVariant';

export interface IMassiveDbConnection {
  session: IMassiveSession;
  getPathTo(mObject: string): MassiveLocation;
  fullObject(path: IMassiveLocation): IMassiveObject;
  wangle(fullMObject: IMassiveObject): MassiveLocation;

  callMassiveMethod(
    path: IMassiveLocation,
    wangleLocation: IMassiveLocation,
    inParameters: IMassiveVariant[],
    outParameters: IMassiveVariant[],
    errors: string[]
  ): number;
}
